### PR TITLE
connects falkonry/tracker#240 

### PR DESCRIPTION
**Pull request by Jaipreet95**
Ported to Python3,Rectified test cases

### CODE REVIEW

File: test/test_TestLiveDatastream.py [hunk 1]
```diff
@@ -35,11 +37,11 @@ def test_turn_datstream_on_off(self):
                 # self.assertEqual(str(listAssessment[0]['live']), 'OFF', 'Cannot turn off live mornitoring')
 
             except Exception as e:
-                print(e.message)
```
Comment (phagunbaya) [inline test/test_TestLiveDatastream.py]: any specific reason for this change ?

File: test/test_TestLiveDatastream.py [hunk 2]
```diff
@@ -5,7 +5,9 @@
 
 host          = os.environ['FALKONRY_HOST_URL']  # host url
 token         = os.environ['FALKONRY_TOKEN']     # auth token
-datastream_id = 'tkp9lrh8vhbknj'                 # datastream id
+datastream_id = 'tkp9lrh8vhbknj'                 # datastream
+###Uncomment and set according to your environment for development
```
Comment (phagunbaya) [inline test/test_TestLiveDatastream.py]: Not required.

File: test/test_TestGetFacts.py [hunk 3]
```diff
@@ -7,6 +7,9 @@
 token       = os.environ['FALKONRY_TOKEN']     # auth token
 assessment  = 'r2h27kn82dvrvy'                 # assessment id
 assessmentB = 'r7l9tddmngbbl6'
+###Uncomment and set according to your environment for development
```
Comment (phagunbaya) [inline test/test_TestGetFacts.py]: not required

File: test/test_TestEntityMeta.py [hunk 4]
```diff
@@ -103,21 +103,21 @@ def test_get_entity_meta(self):
...
             self.assertEqual(isinstance(getEntityMetaResp, Schemas.EntityMeta), True, 'Invalid entityMeta object after creation')
-            self.assertEqual(isinstance(getEntityMetaResp.get_id(), unicode), True, 'Invalid id of entityMeta after creation')
+            self.assertEqual(isinstance(getEntityMetaResp.get_id(), str), True, 'Invalid id of entityMeta after creation')
             self.assertEqual(getEntityMetaResp.get_label(), 'testName', 'Invalid label of entityMeta after creation')
             self.assertEqual(getEntityMetaResp.get_path(), 'root/path', 'Invalid path of entityMeta after creation')
             self.assertEqual(getEntityMetaResp.get_sourceId(), 'testId', 'Invalid sourceId of entityMeta after creation')
 
         except Exception as e:
-            print(e.message)
+            print(e.message if hasattr(e, 'message') else e)
```
Comment (phagunbaya) [inline test/test_TestEntityMeta.py]: Should do same thing at other places

File: test/Test.sh [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
-python TestAssessment.py
-python TestAddData.py
-python TestAddDataFromStream.py
-python TestAddFacts.py
-python TestAddFactsFromStream.py
-python TestGetFacts.py
-python TestLiveDatastream.py
-python TestAssessmentGetOutput.py
\ No newline at end of file
+python test_TestEntityModel.py
```
Comment (phagunbaya) [inline test/Test.sh]: Any reason for changing the names ?

File: setup.py [hunk 6]
```diff
@@ -25,7 +25,7 @@
     tests_require=tests_require,
     setup_requires=setup_requires,
     extras_require={
-        'test': ['unittest', 'io'],
+        'test': ['unittest', 'io', 'pytest'],
```
Comment (phagunbaya) [inline setup.py]: are we using any class from `pytest` ?

File: test/Test.sh (hunk 5 above)
Comment (Jaipreet95) [reply test/Test.sh]: For pytest to find the unitTests and give more verbose output

File: test/test_TestGetFacts.py (hunk 3 above)
Comment (Jaipreet95) [reply test/test_TestGetFacts.py]: Okay

File: test/Test.sh (hunk 5 above)
Comment (phagunbaya) [reply test/Test.sh]: But we are not using pytest either on shippable.

**Comment by Jaipreet95** [comment]
@phagunbaya 
I just wanted to check whether my test fixes were working properly on the deployment or not
There are other changes as well which were done in correspondance with falkonry-cli 
Kindly wait for the Code review
Will let you know about the same when its completely done

File: falkonryclient/helper/utils.py [hunk 7]
```diff
@@ -9,11 +9,60 @@
...
 import string
 import random
-
+import json
 
 def random_string(size=6):
     return ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(size))
 
+
+def exceptionResponseHandler(exceptionObj):
```
Comment (phagunbaya) [inline falkonryclient/helper/utils.py]: should be named as `exception_response_handler`

Comment (Jaipreet95) [reply falkonryclient/helper/utils.py]: Addressed

**Pull request merged**